File: src/main/java/main/json/NumberJson.java
package main.json;

import org.jetbrains.annotations.NotNull;

/**
 *
 * @author Bunnyspa
 */
public class NumberJson implements Json {

    @NotNull
    private final String data;

    public NumberJson(@NotNull String data) {
        this.data = data.trim();
    }

    public int getInteger() {
        return Integer.parseInt(data);
    }

    public double getDouble() {
        return Double.parseDouble(data);
    }

    @Override
    public int getType() {
        return Json.NUMBER;
    }

    @NotNull
    @Override
    public String toString() {
        return data;
    }
}

File: src/main/java/main/json/NumberJson.kt
package main.json

/**
 *
 * @author Bunnyspa
 */
class NumberJson(data: String) : Json {
    private val data: String = data.trim { it <= ' ' }
    val integer: Int
        get() = data.toInt()
    val double: Double
        get() = data.toDouble()
    override val type: Int
        get() = Json.NUMBER

    override fun toString(): String {
        return data
    }

}